src/services/User.service.ts

```diff
@@ -5,8 +5,8 @@ import AppDataSource from "../config/ormconfig";
 export class UserService {
   private userRepository: Repository<User>;
 
-  constructor() {
-    this.userRepository = AppDataSource.getRepository(User);
+  constructor(dataSource: typeof AppDataSource) {
```

Comment:
```suggestion
  constructor() {
```

Why not using the imported `DataSource` ?
If there's a reason feel free to tell me for considering ur proposal

Reply:
Using **AppDataSource** directly in the **UserService** constructor is convenient but has drawbacks as your app grows. I'm recommending, passing **dataSource** as a parameter for better:  

- **Dependency Injection:** Makes your code modular and testable.  
- **Testability:** Easier to mock **dataSource** in unit tests.  
- **Flexibility:** Allows different data sources for different environments when needed.  
- **Decoupling:** Reduces dependency on **AppDataSource**, making future changes easier.

**NOTE: I can revert back to your original implementation you if you prefer it that way. Just let me know.**

Reply:
Ok, Great explanation. We are going to refactor.

Reply:
Other changes have been pushed. You can review them and approve when you're satisfied with the PR. Thank you for giving me this opportunity to contribute to this project.